fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/fragments/event/creation/ElectionSetupFragment.java

```diff
@@ -0,0 +1,202 @@
+            else if (!ballotOptions.get(ballotIndex).isEmpty() && editable.toString().isEmpty())  numberBallotOptions--;
+            // Either way, we save our change in the corresponding index
+            ballotOptions.set(ballotIndex, editable.toString());
+        }
+    }
+
+    public static ElectionSetupFragment newInstance() {
+        return new ElectionSetupFragment();
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(
+            @NonNull LayoutInflater inflater,
+            @Nullable ViewGroup container,
+            @Nullable Bundle savedInstanceState) {
+
+        mSetupElectionFragBinding =
+                FragmentSetupElectionEventBinding.inflate(inflater, container, false);
+
+        LaoDetailViewModel laoDetailViewModel = LaoDetailActivity.obtainViewModel(getActivity());
+
+        setDateAndTimeView(mSetupElectionFragBinding.getRoot(), this, getFragmentManager());
+        addDateAndTimeListener(submitTextWatcher);
+
+        Button cancelButton = mSetupElectionFragBinding.electionCancelButton;
+
+        ballotOptions = new ArrayList<>();
+
+        submitButton = mSetupElectionFragBinding.electionSubmitButton;
+        electionNameText = mSetupElectionFragBinding.electionSetupTitle;
+        electionQuestionText = mSetupElectionFragBinding.electionQuestion;
+
+        //Add text watchers on the fields that need to be filled
+        electionQuestionText.addTextChangedListener(submitTextWatcher);
+        electionNameText.addTextChangedListener(submitTextWatcher);
+
+        // Set up the basic fields for ballot options, with at least two options
+        initNewBallotOptionsField();
+
+        //When the button is clicked, add a new ballot option
+       FloatingActionButton addBallotOptionButton = mSetupElectionFragBinding.addBallotOption;
+        addBallotOptionButton.setOnClickListener(
+                v -> addBallotOption());
+
+
+        // Set the text widget in layout to current LAO name
+        TextView laoNameTextView = mSetupElectionFragBinding.electionSetupLaoName;
+        laoNameTextView.setText(laoDetailViewModel.getCurrentLaoName().getValue());
+
+        //Set dropdown spinner as a list of string (from enum of votingmethods)
+       Spinner spinner = mSetupElectionFragBinding.electionSetupSpinner;
+        String[] items = Arrays.stream(votingMethods.values()).map(votingMethods::toString).toArray(String[]::new);
+        ArrayAdapter<String> adapter = new ArrayAdapter<>(getActivity(),
+                android.R.layout.simple_spinner_item, items);
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinner.setAdapter(adapter);
+        spinner.setOnItemSelectedListener(this);
+
+
+        //On click, submit button creates a new election event and launches the election UI for organizer
+       submitButton.setOnClickListener(
+                v -> {
```

Comment:
I'd suggest disabling the submit button here to prevent multiple clicks.

Reply:
good idea !